src/map/battle.cpp

```diff
@@ -8348,7 +8348,7 @@ int64 battle_calc_return_damage(struct block_list* bl, struct block_list *src, i
 			rdamage = damage * sc->data[SC_MAXPAIN]->val1 * 10 / 100;
 	}
 
-	return cap_value(min(rdamage,max_damage),INT_MIN,INT_MAX);
+	return cap_value(i64min(rdamage,max_damage),INT_MIN,INT_MAX);
```

Comment:
Do we need to adjust INT_MIN and INT_MAX to INT64_MIN and INT64_MAX here?

Reply:
So I guess the difference here is that in renewal the reflected damage is capped to the target's max HP (which is INT_MAX) while in pre-renewal that isn't taken into account. It's also something I need to fix in #6717.